packages/blade/src/components/Collapsible/_decisions/decisions.md

```diff
@@ -0,0 +1,186 @@
+
+<Collapsible>
+  <CollapsibleButton>Button</CollapsibleButton>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+
+<Collapsible>
+  <CollapsibleLink>Button</CollapsibleLink>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+```
+
+> **Note**
+>
+> - `Collapsible` will compose (`CollapsibleButton` or `CollapsibleLink`), `CollapsibleBody`. This ensures all interactions, a11y and animations work seamlessly at user end.
+> - `CollapsibleButton` and `CollapsibleLink` will support a subset of relevant props from `Button` and `Link` respectively.
+
+### Collapsible
+
+| Prop              | Type            | Default     | Description                                                        | Required |
+| ----------------- | --------------- | ----------- | ------------------------------------------------------------------ | -------- |
+| children          | `JSX`           | `undefined` | Composes `CollapsibleButton`, `CollapsibleLink`, `CollapsibleBody` | ✅       |
+| direction         | `bottom`, `top` | `bottom`    | Direction in which the content expands                             |          |
+| defaultIsExpanded | `boolean`       | `true`      | Expands the collapsible content by default (uncontrolled)          |          |
+| isExpanded        | `boolean`       | `undefined` | Expands the collapsible content (controlled)                       |          |
+| onChange          | `function`      | `undefined` | Callback for change in collapsible's expanded state                |          |
+
+> **Note**
+>
+> - Also includes layout based styling props
+> - For using collapsible in controlled state, use `isExpanded` coupled with `onChange`
+> - `onChange` callback signature `({ isExpanded }) => {}`, `isExpanded` represents the collapsible's current state
+
+### CollapsibleButton
+
+Built-in trigger for `Collapsible`, provides a subset of `Button` [properties](../../Button/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `variant`
+- `size`
+- `icon`
+- `iconPosition`
+- `isDisabled`
+- `testID`
+- `accessibilityLabel`
+
+Doesn't support ❌
+
+- `isFullWidth` - can visually break UI
+- `onClick`, `isLoading` - using `onClick` will break built-in interactions (unless we provide additional hooks), `onChange` should already provide a way for users to listen-in on `Collapsible` state. This might have usage for async with `isLoading` in the future but for now not exposing the prop
+- `type` - will be `button` always
+
+### CollapsibleLink
+
+Built-in trigger for `Collapsible`, provides a subset of `Link` [properties](../../Link/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `icon`
+- `isDisabled`
+- `size`
+- `accessibilityLabel`
+- `testID`
+
+Doesn't support ❌
+
+- `iconPosition` - will be `left` for passed `icon` since the `CollapsibleLink` renders a rotating chevron icon to its right
```

Comment:
So Link will have chevron icon by default but Button won't? We'll handle the icon to be up or down on Link but not on Button? This seems very confusing for the end user. Can we keep the behaviour consistent? Lets give icon control to the consumer entirely?

Reply:
The caveat for giving control to users would be inability to animate that, the rotating chevron icon is there only for link-button but not for button.

Reply:
I think for the link variant we can provide the rotating chevron (only let users add a left icon if they want). Need to reconfirm with RK once he's back but from the audits I couldn't find cases where a chevron icon won't work.

Reply:
Assessing with design, will finalize based on design research / audits

Reply:
Cool. Rest of the API LGTM